### ISSUE TITLE
NPE in pathfinding

### ISSUE THREAD

**Issue by Danklor** [Shinoow/AbyssalCraft#166]
MC crashed with an NPE in pathfinding as I stepped through a portal from the Abyssal Wasteland to the Overworld.  The attached logs indicate that Pete was trying to attack me.  Upon trying to reload the world MC crashed immediately with the same error.

I have saved the world in case you'd like me to try anything with it.

MC version 1.10.2
Forge 12.18.2.2121
AbyssalCraft 1.9.3.4
...and many other mods listed in the crashlogs.

[crash-2016-11-05_19.54.16-server.txt](https://github.com/Shinoow/AbyssalCraft/files/573585/crash-2016-11-05_19.54.16-server.txt)
[crash-2016-11-05_20.24.24-server.txt](https://github.com/Shinoow/AbyssalCraft/files/573586/crash-2016-11-05_20.24.24-server.txt)

**Comment by Shinoow** [comment]
Do you have any mods installed that alters pathfinding behavior? Because I'm unable to reproduce it with AbyssalCraft and Forge alone. A lot of the lines the stacktrace points to empty lines and opening/closing brackets, so I'm pretty sure there's a mod installed there that's transforming some code in the pathfinding classes through ASM.

**Comment by Danklor** [comment]
I set up an instance with only Forge and AbyssalCraft.  Loading the saved game crashes on every load attempt. (I can re-Pete the crash every time *grin*.)  I've attached the save and another crash log.

[crash-2016-11-06_10.25.34-server.txt](https://github.com/Shinoow/AbyssalCraft/files/573912/crash-2016-11-06_10.25.34-server.txt)
[Test Alpha 008 - Don't be late!.zip](https://github.com/Shinoow/AbyssalCraft/files/573913/Test.Alpha.008.-.Don.t.be.late.zip)

**Comment by Shinoow** [comment]
It seems like entering that world again makes it trigger the crash again. Did you try making another new world with all the mods installed to test if it also happened there? I got that world to stop crashing by having the game regenerate the terrain (eg. deleted the region folder in the save folder for the dimension). In any other cases, I've never managed to trigger that crash in a world.

**Comment by Danklor** [comment]
I fixed my copy of the world with mods by deleting the region files.  It turns out that I was still in the Abyssal Wasteland, but with no mobs in the immediate vicinity since they were wiped with the region files.

I had been playing with this combination of mods for a couple weeks without pathfinding trouble.  I have created 3 new worlds since this "008" test world and have not had any new trouble.  I will continue to test.

**Comment by Danklor** [comment]
I've not duplicated a crash, but I've found a pathfinding glitch with the Depths Ghoul.  I've seen this in a new world in the same place with two different mobs.  When in a 1 block high corner when standing on ooze, they just spin around in circles and can't move.  I've included both saves here.  In both files I am looking at the stuck mob.

The only mods I was using were AbyssalCraft 1.9.3.4, Forge 12.18.2.2121, JEI 3.13.2.359, and JourneyMap 5.3.0b2.

[Abyssal test 011 copy.zip](https://github.com/Shinoow/AbyssalCraft/files/574470/Abyssal.test.011.copy.zip)
[Abyssal test 011.zip](https://github.com/Shinoow/AbyssalCraft/files/574471/Abyssal.test.011.zip)
